[Quant][PT2E] Enable weight scale optimization in QConv PT2E

leslie-fang-intel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107951
* #106958
* #106838
* #106836
* #107565
* __->__ #105996
* #105906
* #105639
* #105456
* #105455
* #104590
* #104588
* #104581
* #104580

**Summary**
After oneDNN 3.1 upgrade, we don't need to do the weight scale reciprocal calculation. So, remove the redundant reciprocal calculation to optimize QConv performance and using IDeep version API to implement it in this PR:

- This QConv implementation expects to work functionally both with current IDeep version and the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565.
- With the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565, the QConv has better performance since the redundant reciprocal calculation are removed.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @gujinghui @PenghuiCheng @jianyuh @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen

leslie-fang-intel committed:
enabel weight scale optimization

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Test Only] Enable weight scale optimization in QConv PT2E"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

jgong5 commented on aten/src/ATen/native/quantized/cpu/qconv.cpp:
```diff
@@ -1473,6 +1473,7 @@ static at::Tensor _quantized_convolution_onednn(
   // Scales of ONEDNN and PyTorch are reciprocal
   const ideep::scale_t& src_scales = ideep::scale_t(1, 1.0 / act_scale);
 
+#if defined(IDEEP_VERSION_MAJOR) && IDEEP_VERSION_MAJOR>=3 && defined(IDEEP_VERSION_REVISION) && IDEEP_VERSION_REVISION == 0
```
What happens with `IDEEP_VERSION_MAJOR < 3`?

leslie-fang-intel replied:
Thanks for the comment. Add the check failure if `IDEEP_VERSION_MAJOR < 3`.

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable weight scale optimization in QConv PT2E"


**Summary**
After oneDNN 3.1 upgrade, we don't need do the weight scale reciprocal calculation. So, remove the duplicated reciprocal calculation to optimize QConv performance.
Change to new IDeep Commit after IDeep PR: https://github.com/intel/ideep/pull/222 landed.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable weight scale optimization in QConv PT2E"


**Summary**
After oneDNN 3.1 upgrade, we don't need do the weight scale reciprocal calculation. So, remove the duplicated reciprocal calculation to optimize QConv performance.
Change to new IDeep Commit after IDeep PR: https://github.com/intel/ideep/pull/222 landed.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable weight scale optimization in QConv PT2E"


**Summary**
After oneDNN 3.1 upgrade, we don't need do the weight scale reciprocal calculation. So, remove the duplicated reciprocal calculation to optimize QConv performance.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable weight scale optimization in QConv PT2E"


**Summary**
After oneDNN 3.1 upgrade, we don't need do the weight scale reciprocal calculation. So, remove the duplicated reciprocal calculation to optimize QConv performance.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable weight scale optimization in QConv PT2E"


**Summary**
After oneDNN 3.1 upgrade, we don't need to do the weight scale reciprocal calculation. So, remove the redundant reciprocal calculation to optimize QConv performance and using IDeep version API to implement it in this PR:

- This QConv implementation expects to work functionally both with current IDeep version and the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565.
- With the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565, the QConv has better performance since the redundant reciprocal calculation are removed.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable weight scale optimization in QConv PT2E"


**Summary**
After oneDNN 3.1 upgrade, we don't need to do the weight scale reciprocal calculation. So, remove the redundant reciprocal calculation to optimize QConv performance and using IDeep version API to implement it in this PR:

- This QConv implementation expects to work functionally both with current IDeep version and the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565.
- With the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565, the QConv has better performance since the redundant reciprocal calculation are removed.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable weight scale optimization in QConv PT2E"


**Summary**
After oneDNN 3.1 upgrade, we don't need to do the weight scale reciprocal calculation. So, remove the redundant reciprocal calculation to optimize QConv performance and using IDeep version API to implement it in this PR:

- This QConv implementation expects to work functionally both with current IDeep version and the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565.
- With the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565, the QConv has better performance since the redundant reciprocal calculation are removed.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable weight scale optimization in QConv PT2E"


**Summary**
After oneDNN 3.1 upgrade, we don't need to do the weight scale reciprocal calculation. So, remove the redundant reciprocal calculation to optimize QConv performance and using IDeep version API to implement it in this PR:

- This QConv implementation expects to work functionally both with current IDeep version and the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565.
- With the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565, the QConv has better performance since the redundant reciprocal calculation are removed.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable weight scale optimization in QConv PT2E"


**Summary**
After oneDNN 3.1 upgrade, we don't need to do the weight scale reciprocal calculation. So, remove the redundant reciprocal calculation to optimize QConv performance and using IDeep version API to implement it in this PR:

- This QConv implementation expects to work functionally both with current IDeep version and the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565.
- With the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565, the QConv has better performance since the redundant reciprocal calculation are removed.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable weight scale optimization in QConv PT2E"


**Summary**
After oneDNN 3.1 upgrade, we don't need to do the weight scale reciprocal calculation. So, remove the redundant reciprocal calculation to optimize QConv performance and using IDeep version API to implement it in this PR:

- This QConv implementation expects to work functionally both with current IDeep version and the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565.
- With the following IDeep upgrade in PR: https://github.com/pytorch/pytorch/pull/107565, the QConv has better performance since the redundant reciprocal calculation are removed.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 gujinghui PenghuiCheng jianyuh min-jean-cho yanbing-j Guobing-Chen Xia-Weiwen

[ghstack-poisoned]

leslie-fang-intel commented:
@pytorchbot merge
